File: PE1/PE1/Problem16.fs
namespace Program

open System

module Problem16 =

    let toString x =
        x.ToString().ToCharArray()
    
    let Solve = 
        Numerics.BigInteger (Math.Pow(2.0, 1000.0))
        |> toString
        |> Array.map (fun x -> Int32.Parse(x.ToString()))
        |> Array.sum
        
        

File: ProjectEuler/Problems/Problem16.cs
using System;
using System.Linq;
using System.Numerics;

namespace ProjectEuler.Problems
{
    public class Problem16 : Problem
    {
        protected override long DoSolutionSolve()
        {
            return new BigInteger(Math.Pow(2, 1000)).ToString().ToCharArray().Select(x => Convert.ToInt64(x.ToString())).Sum();
        }
    }
}
